pkg/spdk/engine.go

```diff
@@ -122,51 +123,26 @@ func (e *Engine) Create(spdkClient *spdkclient.Client, replicaAddressMap, localR
 		return nil, err
 	}
 
-	nqn := helpertypes.GetNQN(e.Name)
-	port, _, err := superiorPortAllocator.AllocateRange(1)
-	if err != nil {
-		return nil, err
-	}
-	if err := spdkClient.StartExposeBdev(nqn, e.Name, podIP, strconv.Itoa(int(port))); err != nil {
+	if err := e.handleFrontend(spdkClient, superiorPortAllocator); err != nil {
 		return nil, err
 	}
-	e.IP = podIP
-	e.Port = port
-
-	switch e.Frontend {
-	case types.FrontendSPDKTCPBlockdev:
-		initiator, err := nvme.NewInitiator(e.VolumeName, nqn, nvme.HostProc)
-		if err != nil {
-			return nil, err
-		}
-		if err := initiator.Start(podIP, strconv.Itoa(int(port))); err != nil {
-			return nil, err
-		}
-		e.Endpoint = initiator.GetEndpoint()
-	case types.FrontendSPDKTCPNvmf:
-		e.Endpoint = GetNvmfEndpoint(nqn, e.IP, e.Port)
-	case types.FrontendSPDKVoid:
-		logrus.Infof("No frontend specified, will not expose the volume %s", e.VolumeName)
-	default:
-		return nil, fmt.Errorf("unknown frontend type %s", e.Frontend)
-	}
 
 	e.State = types.InstanceStateRunning
 
 	return e.getWithoutLock(), nil
 }
 
-func getBdevNameForReplica(spdkClient *spdkclient.Client, localReplicaBdevMap map[string]string, replicaName, replicaAddress, podIP string) (bdevName string, err error) {
+func getBdevNameForReplica(spdkClient *spdkclient.Client, localReplicaLvsNameMap map[string]string, replicaName, replicaAddress, podIP string) (bdevName string, err error) {
 	replicaIP, replicaPort, err := net.SplitHostPort(replicaAddress)
 	if err != nil {
 		return "", err
 	}
 	if replicaIP == podIP {
-		if localReplicaBdevMap[replicaName] == "" {
-			return "", fmt.Errorf("cannot to find local replica %s from the local replica bdev map", replicaName)
+		if localReplicaLvsNameMap[replicaName] == "" {
+			return "", fmt.Errorf("cannot to find local replica %s from the local replica map", replicaName)
```

Comment:
"failed to ..."

Reply:
I am thinking if we need to differentiate the 2 cases:
1. Some errors fail the get/find. Then we use `failed to ...`
2. There is no such an entry in the map/list. Then we use `cannot find ...`

Reply:
Same here, because we check the words `cannot find ... ` in the longhorn-manager part for now and hould not always use `failed to ...`. 
We can improve the wording in the future.

Reply:
Sorry...? Which one should we use here?

Reply:
How about `cannot find ...`

Reply:
Sure. Then I will follow the convention I mentioned above.